### PR TITLE
chore: add V2 accounts API, restore integration tests

### LINKED ISSUE #816: V2 Accounts API

### PR DESCRIPTION
**Pull request by pscott31**
Closes #816

### CODE REVIEW

File: api/trading_data_v2.go [hunk 1]
```diff
@@ -68,6 +68,65 @@ type tradingDataServiceV2 struct {
 	transfersService          *service.Transfer
 }
 
+func (t *tradingDataServiceV2) ListAccounts(ctx context.Context, req *v2.ListAccountsRequest) (*v2.ListAccountsResponse, error) {
+	if t.accountService == nil {
+		return nil, apiError(codes.Internal, fmt.Errorf("sql balance store not available"))
```
Comment (ettec) [inline api/trading_data_v2.go]: message needs updating

File: api/trading_data_v2.go [hunk 2]
```diff
@@ -68,6 +68,65 @@ type tradingDataServiceV2 struct {
 	transfersService          *service.Transfer
 }
 
+func (t *tradingDataServiceV2) ListAccounts(ctx context.Context, req *v2.ListAccountsRequest) (*v2.ListAccountsResponse, error) {
+	if t.accountService == nil {
```
Comment (ettec) [inline api/trading_data_v2.go:73]: We should add metrics to all V2 apis as per V1

File: api/trading_data_v2.go (hunk 1 above)
Comment (pscott31) [reply api/trading_data_v2.go]: sure thing, done

File: api/trading_data_v2.go (hunk 2 above)
Comment (pscott31) [reply api/trading_data_v2.go:73]: good point, done

**Review by ettec — DISMISSED** [review]

**Review by ettec — APPROVED** [review]

**Pull request merged**